test/integration/consul-container/libs/node/consul-container.go

```diff
@@ -112,25 +117,37 @@ func NewConsulContainer(ctx context.Context, config Config) (Node, error) {
 
 	return &consulContainerNode{
 		config:    config,
-		container: container,
+		pod:       podContainer,
+		container: consulContainer,
 		ip:        ip,
 		port:      mappedPort.Int(),
 		client:    apiClient,
 		ctx:       ctx,
-		req:       req,
+		podReq:    podReq,
+		consulReq: consulReq,
 		dataDir:   tmpDirData,
 	}, nil
 }
 
-func newContainerRequest(config Config, name, configFile, dataDir, license string) testcontainers.ContainerRequest {
+const pauseImage = "k8s.gcr.io/pause:3.3"
+
+func newContainerRequest(config Config, name, configFile, dataDir, license string) (podRequest, consulRequest testcontainers.ContainerRequest) {
 	skipReaper := isRYUKDisabled()
 
-	return testcontainers.ContainerRequest{
-		Image:        consulImage + ":" + config.Version,
-		ExposedPorts: []string{"8500/tcp"},
-		WaitingFor:   wait.ForLog(bootLogLine).WithStartupTimeout(10 * time.Second),
+	pod := testcontainers.ContainerRequest{
```

Comment:
hm, there could be a `newPod` func that wraps around to create a `pod`. Same for `app`. That way we could mix and match down the line.